File: OBI/2022/fase1/chuva.py
N = int(input())

S = int(input())


# aplica o casting para inteiro para cada item do input splitado por espaço, e tudo isso é convertido em uma list
chuvas = list(map(int, input().split()))


intervals = 0
for i in range(0, N):
    current_sum = 0
    current = i
    while current < N:
        current_sum += chuvas[current]
        if current_sum == S:
            intervals += 1
        elif current_sum > S:
            break
        current += 1
print(intervals)

File: OBI/2022/fase1/chuva.c
#include <stdio.h>

int main() {
  int N, S;
  scanf("%d %d", &N, &S);
  int chuvas[N];

  for (int i = 0; i < N; i++)
    scanf("%d", &chuvas[i]);

  int current = 0;
  int end = N;
  int intervals = 0;

  for (int i = 0; i < N; i++) {
    int curr_sum = 0;
    current = i;
    while (current < N) {
      curr_sum += chuvas[current];
      if (curr_sum == S)
        intervals++;
      else if (curr_sum > S)
        break;

      current++;
    }
  }

  printf("%d\n", intervals);
}
